Sources/Amplitude/Amplitude.m

```diff
@@ -1672,6 +1672,10 @@ - (NSString *)getDeviceId {
     return self.deviceId;
 }
 
+- (NSString *)getDeviceIdSync {
+    return [self initializeDeviceId];
```

Comment:
I think I need some mutual exclusion here since the async initialization here runs this same function on a different queue/thread.